src/Components/Server/src/ComponentHub.cs

```diff
@@ -129,10 +131,56 @@ public async Task<bool> ConnectCircuit(string circuitId)
         /// </summary>
         public void BeginInvokeDotNetFromJS(string callId, string assemblyName, string methodIdentifier, long dotNetObjectId, string argsJson)
         {
-            Log.BeginInvokeDotNet(_logger, callId, assemblyName, methodIdentifier, dotNetObjectId);   
-            EnsureCircuitHost().BeginInvokeDotNetFromJS(callId, assemblyName, methodIdentifier, dotNetObjectId, argsJson);
+            Log.BeginInvokeDotNet(_logger, callId, assemblyName, methodIdentifier, dotNetObjectId);
+            var _ = EnsureCircuitHost().BeginInvokeDotNetFromJS(callId, assemblyName, methodIdentifier, dotNetObjectId, argsJson);
         }
 
+        public void EndInvokeDotNetFromJS(string arguments)
+        {
+            var _ = EnsureCircuitHost().EndInvokeDotNetFromJS(arguments);
+        }
+
+        public void DispatchBrowserEvent(string args)
+        {
+            try
+            {
+                var document = JsonDocument.Parse(args);
+                if (document.RootElement.ValueKind != JsonValueKind.Array)
+                {
+                    // Log not array
+                    return;
+                }
+                var length = document.RootElement.GetArrayLength();
+                if (length != 2)
+                {
+                    // Log invalid length
+                }
+                RendererRegistryEventDispatcher.BrowserEventDescriptor eventDescriptor = null;
+                string eventArgsJson = null;
+                foreach (var element in document.RootElement.EnumerateArray())
+                {
+                    if (eventDescriptor == null)
+                    {
+                        eventDescriptor = JsonSerializer.Deserialize<RendererRegistryEventDispatcher.BrowserEventDescriptor>(
+                            element.GetRawText(),
+                            JsonSerializerOptionsProvider.Options);
+                    }
+                    else
+                    {
+                        eventArgsJson = element.GetString();
+                    }
+                }
+                var _ = EnsureCircuitHost().DispatchEvent(eventDescriptor, eventArgsJson);
```

Comment:
We need a way to differentiate between what it is essentially well-formed bad input (like an event for a non-existing handler) and exceptions in user code (Exceptions thrown inside developer's event handling code).